app/code/Magento/Cms/Controller/Adminhtml/Wysiwyg/Images/OnInsert.php

```diff
@@ -34,29 +43,18 @@ public function __construct(
      */
     public function execute()
     {
-        $imagesHelper = $this->_objectManager->get(\Magento\Cms\Helper\Wysiwyg\Images::class);
-        $request = $this->getRequest();
-
-        $storeId = $request->getParam('store');
-
-        $filename = $request->getParam('filename');
-        $filename = $imagesHelper->idDecode($filename);
-
-        $asIs = $request->getParam('as_is');
-
-        $forceStaticPath = $request->getParam('force_static_path');
-
-        $this->_objectManager->get(\Magento\Catalog\Helper\Data::class)->setStoreId($storeId);
-        $imagesHelper->setStoreId($storeId);
-
-        if ($forceStaticPath) {
-            $image = parse_url($imagesHelper->getCurrentUrl() . $filename, PHP_URL_PATH);
-        } else {
-            $image = $imagesHelper->getImageHtmlDeclaration($filename, $asIs);
-        }
+        $data = $this->getRequest()->getParams();
 
         /** @var \Magento\Framework\Controller\Result\Raw $resultRaw */
         $resultRaw = $this->resultRawFactory->create();
-        return $resultRaw->setContents($image);
+
+        return $resultRaw->setContents(
+            $this->getInsertImageContent->execute(
+                $data['filename'],
```

Comment:
I think it would be good to add the parameter names to a constant (constant names should help with the readability of the code)